From 6781213261c36f4f6c417b06da630a49484faab4 Mon Sep 17 00:00:00 2001
From: Lars Uebernickel <lars.uebernickel@canonical.com>
Date: Mon, 10 Feb 2014 18:13:56 +0100
Subject: [PATCH] Don't use gtk_application_set_accels_for_action()

Gtk 3.10 doesn't have this function. Change that to use
gtk_application_add_accelerator. Also add the accelerators to menus.xml
so that they show up in the UI. (It should be enough to only add them to
menus.xml, but gtk doesn't seem to pick those up. It's not worth fixing
that bug as the accelerator handling is revamped in 3.12 and the this
workaround works fine.)
---
 shell/ev-application.c |  2 +-
 shell/menus.ui         | 22 ++++++++++++++++++++++
 2 files changed, 23 insertions(+), 1 deletion(-)

Index: evince-3.10.3/shell/ev-application.c
===================================================================
--- evince-3.10.3.orig/shell/ev-application.c	2014-04-29 16:09:07.319931409 +0200
+++ evince-3.10.3/shell/ev-application.c	2014-04-29 16:09:33.031931585 +0200
@@ -1193,7 +1193,17 @@
         it = action_accels;
         while (it[0])
           {
-            gtk_application_set_accels_for_action (GTK_APPLICATION (application), it[0], &it[1]);
+            gchar *action;
+            GVariant *target;
+
+            if (g_action_parse_detailed_name (it[0], &action, &target, NULL))
+              {
+                gtk_application_add_accelerator (GTK_APPLICATION (application), it[1], action, target);
+
+                g_free (action);
+                if (target)
+                  g_variant_unref (target);
+              }
             it += g_strv_length ((gchar **) it) + 1;
           }
 }
Index: evince-3.10.3/shell/menus.ui
===================================================================
--- evince-3.10.3.orig/shell/menus.ui	2014-04-29 16:09:07.319931409 +0200
+++ evince-3.10.3/shell/menus.ui	2014-04-29 16:09:07.311931409 +0200
@@ -23,6 +23,7 @@
       <item>
         <attribute name="label" translatable="yes">_Open</attribute>
         <attribute name="action">app.open</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;O</attribute>
       </item>
     </section>
     <section>
@@ -33,6 +34,7 @@
       <item>
         <attribute name="label" translatable="yes">_Help</attribute>
         <attribute name="action">app.help</attribute>
+        <attribute name="accel">F1</attribute>
       </item>
     </section>
   </menu>
@@ -52,36 +54,43 @@
       <item>
         <attribute name="label" translatable="yes">Side _Pane</attribute>
         <attribute name="action">win.show-side-pane</attribute>
+        <attribute name="accel">F9</attribute>
       </item>
     </section>
     <section>
       <item>
         <attribute name="label" translatable="yes">_Fullscreen</attribute>
         <attribute name="action">win.fullscreen</attribute>
+        <attribute name="accel">F11</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">Pre_sentation</attribute>
         <attribute name="action">win.presentation</attribute>
+        <attribute name="accel">F5</attribute>
       </item>
     </section>
     <section>
       <item>
         <attribute name="label" translatable="yes">Rotate _Left</attribute>
         <attribute name="action">win.rotate-left</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;Left</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">Rotate _Right</attribute>
         <attribute name="action">win.rotate-right</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;Right</attribute>
       </item>
     </section>
     <section>
       <item>
         <attribute name="label" translatable="yes">Zoom _In</attribute>
         <attribute name="action">win.zoom-in</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;plus</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">Zoom _Out</attribute>
         <attribute name="action">win.zoom-out</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;minus</attribute>
       </item>
     </section>
     <section>
@@ -92,12 +101,14 @@
       <item>
         <attribute name="label" translatable="yes">_Inverted Colors</attribute>
         <attribute name="action">win.inverted-colors</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;I</attribute>
       </item>
     </section>
     <section>
       <item>
         <attribute name="label" translatable="yes">_Reload</attribute>
         <attribute name="action">win.reload</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;R</attribute>
       </item>
     </section>
   </menu>
@@ -107,10 +118,12 @@
       <item>
         <attribute name="label" translatable="yes">_Open…</attribute>
         <attribute name="action">app.open</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;O</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">Op_en a Copy</attribute>
         <attribute name="action">win.open-copy</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;N</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">_Recent</attribute>
@@ -120,6 +133,7 @@
       <item>
         <attribute name="label" translatable="yes">_Save a Copy…</attribute>
         <attribute name="action">win.save-copy</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;S</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">Send _To…</attribute>
@@ -132,22 +146,26 @@
       <item>
         <attribute name="label" translatable="yes">_Print…</attribute>
         <attribute name="action">win.print</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;P</attribute>
       </item>
     </section>
     <section>
       <item>
         <attribute name="label" translatable="yes">P_roperties…</attribute>
         <attribute name="action">win.show-properties</attribute>
+        <attribute name="accel">&lt;Alt&gt;Return</attribute>
       </item>
     </section>
     <section>
       <item>
         <attribute name="label" translatable="yes">_Copy</attribute>
         <attribute name="action">win.copy</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;C</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">Select _All</attribute>
         <attribute name="action">win.select-all</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;A</attribute>
       </item>
     </section>
     <section>
@@ -160,22 +178,26 @@
       <item>
         <attribute name="label" translatable="yes">_First Page</attribute>
         <attribute name="action">win.go-first-page</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;Home</attribute>
       </item>
       <item>
         <attribute name="label" translatable="yes">_Last Page</attribute>
         <attribute name="action">win.go-last-page</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;End</attribute>
       </item>
     </section>
     <section id="bookmarks">
       <item>
         <attribute name="label" translatable="yes">_Add Bookmark</attribute>
         <attribute name="action">win.add-bookmark</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;D</attribute>
       </item>
     </section>
     <section>
       <item>
         <attribute name="label" translatable="yes">_Close</attribute>
         <attribute name="action">win.close</attribute>
+        <attribute name="accel">&lt;Ctrl&gt;W</attribute>
       </item>
     </section>
     <section>
